### ISSUE TITLE
Cannot launchpad

### ISSUE THREAD

**Issue by agrignard** [gama-platform/gama.experimental#15]
I am trying to make working the Launchpad feature developped during the ast Coding CAmp however my GAMA plugins Knowledge is a bit rusted. I put it in the ummisco.gama.feature, then it compiles well however when I open my model (https://github.com/gama-platform/gama.experimental/tree/master/mit.gaml.extension.launchpad/models) there is one action that is not recognize anymore. I didn't remember touching anything in the code, I guess something is outdated with the GAMA core. 

@AlexisDrogoul and/or @hqnghi88 can I ask you some help on this?

<img width="1920" alt="screen shot 2018-12-18 at 16 08 58" src="https://user-images.githubusercontent.com/3928502/50183128-2dbf8700-02df-11e9-80fa-e8c840dcdfe5.png">

**Comment by benoitgaudou** [comment]
Hi Arnaud,

I guess you do have linked your plugin/feature with the GAMA you are
running.
Has you added the plugin to the feature
:  ummisco.gama.feature.core.extensions ?
Or have you added your feature to the gama.runtime.product
of ummisco.gama.product ?

Cheers

Benoit






Le mer. 19 déc. 2018 à 04:10, Arnaud Grignard <notifications@github.com> a
écrit :

**Comment by hqnghi88** [comment]
My apology Arnaud, it seems some code had been missed commit. I have done a full review on history. Can you pull, retry and tell me the next issue?

**Comment by agrignard** [comment]
@hqnghi88 yes it works much better with the initial code ;-)

I knew I didn't change anything and I was not able to understand what has changed.

Anyway thanks for the revert, let's keep in touch I am realiving this feature so I might have new specific question

**Comment by agrignard** [comment]
I have a new issue while trying to install the launchpad plugin on the official 1.8 release.

The plugins is well installed howver I get a compiel error when writting this


`````
event "pad_down" type: "launchpad" action: updateGrid;
``````

Saying
````
No event can be triggered for 'pad_down'. Acceptable values are
   [mouse_exit, mouse_enter, mouse_move, mouse_down, mouse_up]
   or a character
````

**Comment by agrignard** [comment]
Here is the model that is supposed to work

````
/**
* Name: Launch Pad Event Feature
* Author: Arnaud Grignard 
* Description: Model which shows how to use the event layer to trigger an action with a LaunchPad Novation (This model only work with the launchpad plugins extension available in GAMA 1.7 since January 2018)
* Tags: tangible interface, gui, launchpad
 */
model event_layer_model

global skills:[launchpadskill]
{
	list<string> buttonColors <-["red","orange","brown","yellow","lightyellow","green","darkgreen","black"];
	map<string,string> function_map <-["UP"::buttonColors[0],"DOWN"::buttonColors[1],"LEFT"::buttonColors[2],"RIGHT"::buttonColors[3],"SESSION"::buttonColors[4],"USER_1"::buttonColors[5],"USER_2"::buttonColors[6],"MIXER"::buttonColors[7]];
	init{
	  do resetPad;
	  do setButtonLight colors:buttonColors;	
	}
	
	action updateGrid
	{   
		if(function_map.keys contains buttonPressed and buttonPressed != "MIXER"){
		    ask cell[ int(padPressed.y *8 + padPressed.x)]{color <- rgb(function_map[buttonPressed]);}
		    do setPadLight color:function_map[buttonPressed];
		}
		if(buttonPressed = "MIXER"){
			ask cell[ int(padPressed.y *8 + padPressed.x)]{color <- #white;}
		}			
		if(buttonPressed="ARM"){
			do resetPad;
			do setButtonLight colors:buttonColors;	
			ask cell{
				color<-#white;
			}
		}
		do updateDisplay;
	}	
}

grid cell width: 8 height: 8;

experiment Displays type: gui
{
	output
	{
		display View_change_color 
		{
			grid cell lines: #black;
			event "pad_down" type: "launchpad" action: updateGrid;
		}
	}
}
````
